### PR TITLE
[VMware] Add support for VMware 8.0u2 (8.0.2.x) and 8.0u3 (8.0.3.x)

### PR DESCRIPTION
**Pull request by sureshanaparti**
### Description

This PR adds support for VMware 8.0u2 (8.0.2.x) and 8.0u3 (8.0.3.x).

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)
- [ ] build/CI
- [ ] test (unit or integration test code)

#### Feature/Enhancement Scale

- [x] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial

### CODE REVIEW

**Comment by sureshanaparti** [comment]
@blueorangutan package

File: engine/schema/src/main/resources/META-INF/db/schema-41910to42000.sql [hunk 1]
```diff
@@ -224,3 +224,9 @@ CALL `cloud`.`IDEMPOTENT_MODIFY_COLUMN_CHAR_SET`('vpc', 'display_text', 'VARCHAR
 CALL `cloud`.`IDEMPOTENT_MODIFY_COLUMN_CHAR_SET`('vpc_offerings', 'name', 'VARCHAR(255)', 'DEFAULT NULL COMMENT \'vpc offering name\'');
 CALL `cloud`.`IDEMPOTENT_MODIFY_COLUMN_CHAR_SET`('vpc_offerings', 'unique_name', 'VARCHAR(64)', 'DEFAULT NULL COMMENT \'unique name of the vpc offering\'');
 CALL `cloud`.`IDEMPOTENT_MODIFY_COLUMN_CHAR_SET`('vpc_offerings', 'display_text', 'VARCHAR(255)', 'DEFAULT NULL COMMENT \'display text\'');
+
+-- Add support for VMware 8.0u2 (8.0.2.x) and 8.0u3 (8.0.3.x)
```
Comment (weizhouapache) [inline engine/schema/src/main/resources/META-INF/db/schema-41910to42000.sql:228]: @sureshanaparti 
have you checked if there are new supported guest OS in 8.0.2.x and 8.0.3.x ?

Comment (sureshanaparti) [reply engine/schema/src/main/resources/META-INF/db/schema-41910to42000.sql:228]: no new supported guest OSes @weizhouapache

**Review by rohityadavcloud — APPROVED** [review]

**Comment by sureshanaparti** [comment]
@blueorangutan package

**Comment by sureshanaparti** [comment]
@blueorangutan package

File: tools/appliance/build.sh [hunk 2]
```diff
@@ -235,7 +235,7 @@ function stage_vmx() {
 displayname = "${1}"
 annotation = "${1}"
 guestos = "otherlinux-64"
-virtualHW.version = "11"
+virtualHW.version = "13"
```
Comment (rohityadavcloud) [inline tools/appliance/build.sh:238]: will this be compatible with ESXi 6.x and 7.x @sureshanaparti cc @weizhouapache ?

**Review by rohityadavcloud — APPROVED** [review]

**Comment by rohityadavcloud** [comment]
systemvmtemplate build kicked, template available in next 30mins at https://build.yadav.cloud/systemvm/

**Comment by rohityadavcloud** [comment]
@blueorangutan package

**Comment by rohityadavcloud** [comment]
@blueorangutan package

**Comment by rohityadavcloud** [comment]
Systemvmtemplate build OK https://build.yadav.cloud/systemvm/9591-x86_64/

File: tools/appliance/build.sh (hunk 2 above)
Comment (sureshanaparti) [reply tools/appliance/build.sh:238]: yes, compatible @rohityadavcloud 
https://docs.vmware.com/en/VMware-vSphere/8.0/vsphere-vm-administration/GUID-64D4B1C9-CD5D-4C68-8B50-585F6A87EBA0.html)

File: tools/appliance/systemvmtemplate/scripts/configure_grub.sh [hunk 3]
```diff
@@ -34,7 +34,7 @@ function configure_grub() {
 GRUB_DEFAULT=0
 GRUB_TIMEOUT=0
 GRUB_DISTRIBUTOR=Debian
-GRUB_CMDLINE_LINUX_DEFAULT="quiet"
+GRUB_CMDLINE_LINUX_DEFAULT="quiet fsck.mode=force fsck.repair=yes"
```
Comment (sureshanaparti) [inline tools/appliance/systemvmtemplate/scripts/configure_grub.sh:37]: these systemvm templates changes are merged with the PR https://github.com/apache/cloudstack/pull/9625.

**Comment by sureshanaparti** [comment]
@blueorangutan package

**Review by vishesh92 — APPROVED** [review]
clgtm. Can be merged once the test runs are complete.

**Comment by sureshanaparti** [comment]
@blueorangutan help

**Review by Pearl1594 — APPROVED** [review]
clgtm. Changes made to bo as well

**Review by weizhouapache — APPROVED** [review]
code lgtm

checked with Suresh on the guest os identifiers
tested new systemvm template on vmware 80u3 env

**Comment by DaanHoogland** [comment]
@blueorangutan test ol8 vmware-80u2

**Comment by rohityadavcloud** [comment]
Smoketests wouldn't do justice as new systemvmtemplate hasn't been uploaded yet. Merging this based on code reviews and manual tests, will kick tests on health check PR - after building/replacing the new systemvmtemplates.

**Pull request merged**